internal/services/appservice/linux_function_app_resource.go

```diff
@@ -126,11 +132,22 @@ func (r LinuxFunctionAppResource) Arguments() map[string]*pluginsdk.Schema {
 			Type:     pluginsdk.TypeBool,
 			Optional: true,
 			Default:  false,
-			ExactlyOneOf: []string{
-				"storage_uses_managed_identity",
+			ConflictsWith: []string{
 				"storage_account_access_key",
+				"storage_key_vault_secret_id",
 			},
-			Description: "Should the Function App use its Managed Identity to access storage",
+			Description: "Should the Function App use its Managed Identity to access storage?",
+		},
+
+		"storage_key_vault_secret_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: kvValidate.NestedItemId,
+			ExactlyOneOf: []string{
+				"storage_account_name",
+				"storage_key_vault_secret_id",
+			},
+			Description: "The Key Vault Secret ID, including version, that contains the Connection String to connect to the storage account for this Function App.",
```

Comment:
If you decide to rename the argument, I'd propose to update the description along the lines of: `A Key Vault reference string pointing to a secret that contains the Connection String to connect to the storage account for this Function App.` If possible maybe link directly to the MS docs on the reference string.